### DISCUSSION TITLE
Enable experimental features

### DISCUSSION THREAD

**Discussion by niutech** [pulse-browser/browser#162] [Ideas]
Please enable by default the experimental support for:
- [x] [Compact mode](https://support.mozilla.org/en-US/kb/compact-mode-workaround-firefox) -> `browser.compactmode.show`
- [x] JPEG XL images -> `image.jxl.enabled`
- [ ] inert HTML attribute -> `html5.inert.enabled`
- [ ] password display -> `layout.forms.input-type-show-password-button.enabled`
- [ ] initial-letter CSS property -> `layout.css.initial-letter.enabled` Not implemented: https://bugzilla.mozilla.org/show_bug.cgi?id=1223880
- [x] backdrop-filter CSS property -> `layout.css.backdrop-filter.enabled` Enabled since firefox `103`
- [x] fit-content() CSS function -> `layout.css.fit-content-function.enabled` Enabled since firefox `52` 
- [ ] container query length units -> `layout.css.container-queries.enabled`
- [ ] scroll-linked animations -> `layout.css.scroll-linked-animations.enabled`
- [ ] :has() CSS pseudo-class -> `layout.css.has-selector.enabled`
- [x] OffscreenCanvas -> `gfx.offscreencanvas.enabled` Enabled since firefox `105`
- [ ] WebGPU -> `dom.webgpu.enabled`
- [ ] selectAudioOutput() function -> `media.setsinkid.enabled`
- [ ] ~asynchronous SourceBuffer -> `media.mediasource.experimental.enabled`~ not on track to become a standard
- [ ] OpenFont COLRv1 -> `gfx.font_rendering.colr_v1.enabled` Enabled since firefox `107`
- [ ] ClipboardItem and ClipboardRead -> `dom.events.asyncClipboard.clipboardItem` and `dom.events.asyncClipboard.read` (`Clipboard.read()`)
- [ ] HTML Sanitizer API -> `dom.security.sanitizer.enabled ` (partly implemented)
- [ ] Payment Request API -> `dom.payments.request.enabled` + region `dom.payments.request.supportedRegions`
- [ ] Prioritized Task Scheduling API -> `dom.enable_web_task_scheduling`
- [ ] Permissions-Policy header -> `dom.security.featurePolicy.header.enabled`
- [ ] Clear-Site-Data: "cache" header -> `privacy.clearsitedata.cache.enabled `
- [ ] Web console context selector -> `devtools.webconsole.input.context`
- [ ] WebXR -> `dom.vr.webxr.enabled` (WIP: https://bugzilla.mozilla.org/show_bug.cgi?id=1419190)

**Comment by trickypr** [comment]
I have updated your issue to note which ones are already enabled. Some general questions:
- What is "password display"?
- "Web console context selector" appears to be [enabled without any preference.](https://bugzilla.mozilla.org/show_bug.cgi?id=1780156) Am I wrong?

Before enabling anything, I want to make sure that each feature isn't disabled for a security or privacy reason

**Reply by niutech** [reply]
Have a look at [password display](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Experimental_features#toggle_password_display), [execution context selector](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Experimental_features#execution_context_selector), [backdrop-filter](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Experimental_features#backdrop-filter_property), [fit-content()](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Experimental_features#fit-content_function), [JPEG XL](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Experimental_features#jpeg_xl_support), [OffscreenCanvas](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Experimental_features#interface_offscreencanvas) - they say that they are all disabled by default.
Pale Moon has already [enabled JPEG XL support](https://www.palemoon.org/releasenotes.shtml#v31.4.0).

**Reply by trickypr** [reply]
I am not convinced that the "Experimental features" section is up to date.

We already have JPEG XL enabled already.

**Reply by niutech** [reply]
If it is enabled by default, that's awesome! Why not share it to the world? Are there any release notes where it is mentioned?

**Reply by trickypr** [reply]
Errr... We don't have release notes or a change list. Should probably start working on that at some point

**Reply by niutech** [reply]
There is a [changelog](https://github.com/pulse-browser/browser/blob/alpha/CHANGELOG.md), but it is not updated.

**Comment by splatboydev** [comment]
### Async SourceBuffers appear to be non-standard and unsupported as of right now.
![image](https://user-images.githubusercontent.com/64037464/204127657-20b4dc7f-abe1-4ca7-9b87-36a570aea47b.png)
![image](https://user-images.githubusercontent.com/64037464/204127669-d84a764b-e00b-4432-b03a-bd78e8ff5c9d.png)
Taken from [MDN](https://developer.mozilla.org/en-US/docs/Web/API/SourceBuffer/appendBufferAsync)

**Reply by niutech** [reply]
OK, you can ignore it.

**Comment by trickypr** [comment]
I am going to add webxr to the list, because it annoys me slightly that I can't play around with a-frame in Pulse and that I have to use edge.
